### PR TITLE
I1 Item

### PR DESCRIPTION
**Pull request by elliotmitchum**
Refactors the`Item` class and the `ItemType` enum.

Criteria:

- Evaluate against the class diagram
- Evaluate against the code guidelines
- Project builds and runs

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+[*.java]
+indent_style = space
```
Comment (elliotmitchum) [inline .editorconfig]: Test comment

**Comment by elliotmitchum** [comment]
@amrmosit This is ready for review.

**Comment by amrmosit** [comment]
Hey @haynescrossley 
This PR is ready for review. 
Code is still operational. (Checked)

File: src/library/entities/Item.java [hunk 2]
```diff
@@ -1,104 +1,112 @@
...
+        return id;
+    }
+
+
+    public String getTitle() {
+        return title;
+    }
+
+
+    public ItemType getType() {
```
Comment (haynescrossley) [inline src/library/entities/Item.java]: class diagram has this method listed as getItemType(). Please refactor/rename method name.

File: src/library/entities/Item.java [hunk 3]
```diff
@@ -1,104 +1,112 @@
...
+        return state == ItemState.ON_LOAN;
+    }
+
+
+    public boolean isDamaged() {
+        return state == ItemState.DAMAGED;
+    }
+
+
+    public void takeout() {
```
Comment (haynescrossley) [inline src/library/entities/Item.java]: Please refactor/rename with camelCase. Class diagram has takeOut()

File: src/library/entities/Item.java [hunk 4]
```diff
@@ -1,104 +1,112 @@
...
+            state = ItemState.ON_LOAN;
+        }
+        else {
+            String message = String.format("Item: cannot borrow item while item is in state: %s", state);
+            throw new RuntimeException(message);
+        }
+    }
+
+
+    public void takeback(boolean damaged) {
```
Comment (haynescrossley) [inline src/library/entities/Item.java]: Please refactor/rename with camelCase. Class diagram has takeBack()

**Review by haynescrossley — CHANGES_REQUESTED** [review]
@elliotmitchum please check my comments. Only a couple of method names need to be changed to camelcase and one that didnt match the class diagram/definitions. Cheers, Haynes.

**Review by haynescrossley — DISMISSED** [review]
Ready for merge. I'm happy with the changes made.
Changes addressed via commits: 

Rename type to itemType
832113d

Rename takeout to takeOut
bd7a500

Rename takeback to takeBack
275241e

Main has been merged into this branch with the library changes, so this is ready to merge into main @amrmosit.

**Comment by amrmosit** [comment]
@elliotmitchum @haynescrossley 
There are some conflicts to be resolved after merging loan branch

**Review by haynescrossley — APPROVED** [review]
PR has been updated with latest version of Master (i.e., after Loan was merged).

Ready now to merge into main @amrmosit

**Comment by amrmosit** [comment]
i1-item-2 branch has been merged

**Pull request merged**